Library/Homebrew/dev-cmd/pull.rb

```diff
@@ -137,27 +137,33 @@ def pull
         opoo "Current branch is #{branch}: do you need to pull inside master?"
       end
 
-      patch_puller = PatchPuller.new(url, args)
-      patch_puller.fetch_patch
-      patch_changes = files_changed_in_patch(patch_puller.patchpath, tap)
-
-      is_bumpable = patch_changes[:formulae].length == 1 && patch_changes[:others].empty?
-      if do_bump
-        odie "No changed formulae found to bump" if patch_changes[:formulae].empty?
-        if patch_changes[:formulae].length > 1
-          odie "Can only bump one changed formula; bumped #{patch_changes[:formulae]}"
+      merge_commit = merge_commit?(url)
+
+      unless merge_commit
```

Comment:
Maybe this could all be a separate function?

Reply:
Can I do that in a separate PR? The change is already getting quite big, and changing this part is a different part of refactoring. Or should I squash my commits, and make a second separate commit for this part?

Reply:
If you feel strongly you could but I think given the added branching this PR adds it's making `pull` harder to read and splitting into functions would mitigate (and perhaps improve) this. Up to you 👍